File: reference.vb.net/refactoring/refactoring/Refactored/RefactoringUtil.vb
' Copyright (C) 2007  db4objects Inc.  http://www.db4o.com 

Imports Db4objects.Db4o
Imports Db4objects.Db4odoc.Refactoring.Initial

Namespace Db4objects.Db4odoc.Refactoring.Refactored
    Class RefactoringUtil

        Private Const Db4oFileName As String = "reference.db4o"

        Public Shared Sub Main(ByVal args As String())
            MoveValues()
        End Sub

        Public Shared Sub MoveValues()
            Dim container As IObjectContainer = Db4oFactory.OpenFile(Db4oFileName)
            Try
                ' querying for B will bring back B and C values
                Dim result As IObjectSet = container.[Get](New Initial.B())
                For i As Integer = 0 To result.Count - 1
                    Dim b As Initial.B = DirectCast(result(i), Initial.B)
                    Dim d As New D()
                    d.name = b.name
                    d.number = b.number
                    container.Delete(b)
                    container.[Set](d)

                Next
            Finally
                container.Close()
                System.Console.WriteLine("Done")
            End Try
        End Sub
        ' end moveValues


    End Class

End Namespace

File: reference.net/Refactoring/Refactoring/Refactored/RefactoringUtil.cs
/* Copyright (C) 2007  db4objects Inc.  http://www.db4o.com */
using Db4objects.Db4o;
using Db4objects.Db4odoc.Refactoring.Initial;

namespace Db4objects.Db4odoc.Refactoring.Refactored
{
    class RefactoringUtil
    {

        private const string Db4oFileName = "reference.db4o";

        public static void Main(string[] args)
        {
            MoveValues();
        }

        public static void MoveValues()
        {
            IObjectContainer container = Db4oFactory.OpenFile(Db4oFileName);
            try
            {
                // querying for B will bring back B and C values
                IObjectSet result = container.Get(new B());
                for (int i = 0; i < result.Count; i++)
                {
                    B b = (B)result[i];
                    D d = new D();
                    d.name = b.name;
                    d.number = b.number;
                    container.Delete(b);
                    container.Set(d);
                }

            }
            finally
            {
                container.Close();
                System.Console.WriteLine("Done");
            }
        }
        // end moveValues


    }

}
